### ISSUE TITLE
Tensorboard logger: let x-axis be epochs instead of step for training logs

### ISSUE THREAD

**Issue by mpaepper** [PyTorchLightning/pytorch-lightning#4999]
#### What is your question?

Hello PyTorchLightning Community.

We upgraded from version 0.8 to the current version 1.0.8 and then got the error that `training_epoch_end` cannot return any results.

We used that before to aggregate our metrics and then return the averaged logs.

We removed the method and instead are using the self.log_dict now in training_step and it works.

However, on the x-axis there is now a step number which is corresponding to batch update steps and not the epoch number.

What's the recommended way to have it be the epoch number?

#### Code

What we had before:

```
def training_epoch_end(self, outputs):
    avg_loss = torch.stack([x['loss'] for x in outputs]).mean()
    tensorboard_logs = {'0_train/loss': avg_loss}
    for name in self.metrics:
        tensorboard_logs['0_train/{}'.format(name)] = torch.stack([x['metr'][name] for x in outputs]).mean()
    tensorboard_logs['step'] = self.current_epoch  # This was used to log by epoch instead of by step

    return {'loss': avg_loss, 'log': tensorboard_logs}
```

#### What have you tried?

We removed the `training_epoch_end` method and instead in our `training_step` do:

`self.log_dict(metrics_dict, on_step=False, on_epoch=True)`

For validation it still works, so we have the `validation_epoch_end` method and tensorboard logs per epoch and also shows the epoch numbers on the x-axis.

Any help how to have the same behavior in the current version again would be much appreciated.

**Comment by versatran01** [comment]
I have the same question, I was returning a dict of metrics in `validation_epoch_end` before and the model checkpoint will read into it and monitor one of the metric.
And also I log these metrics to tensorboard using epoch instead of step.
Now there is a warning saying `validation_epoch_end` shouldn't return anything, and I have to use `self.log` which has the same problem as above which is I'm not able to specify the x-axis to be epoch rather than step.

**Comment by chenlijun99** [comment]
Has anybody found a workaround to this problem?
